### DISCUSSION TITLE
Expose uuid / ulid

### DISCUSSION THREAD

**Discussion by donferi** [sensedeep/dynamodb-onetable#113] [Q&A] [answered]
Is there a way to import `uuid` and `ulid` helpers?

It looks like I can do something like `table.uuid()` and it will act as `uuid` or `ulid` depending on how you setup the table. But I was wondering if we could just get access to `uuid()` and `ulid()` directly.

```js
import { ulid, uuid } from 'dynamodb-onetable';
```

My use case: I set my table to use `ulid` and use auto-generation for most of my documents, however, I have a few that require `uuid`. Currently I need to generate them with a different library. Would be useful if I could import and just call `uuid()` to generate one.

UPDATE:
Or even better, could we have the models be able to specify `uuid: true` and `ulid: true`? It is a bit weird to use `uuid` and for it to use `ulid` under the hood. That way we don't need to specify `uuid` at the table level.

**Comment by mobsense** [comment]
I think we could export those. Good idea.

You can set {uuid: 'ulid'} in the schema on a per field basis.

**Comment by mobsense** [comment]
Looks like we can do that.

However, the ULID interface is not

    ULID()

ULID is a class that offers other methods to parse ULIDs. So you would invoke it as:

    new ULID().toString()

**Reply by donferi** [reply]
That works 👍

**Reply by donferi** [reply]
Using it on `1.8.0` thanks!! One gotcha is that the `ts` definitions for them are missing, other than that, works, thank you so much!

**Comment by mobsense** [comment, marked as answer]
Will fix the types in 1.8.1
